### PR TITLE
ccmixter connector

### PR DESCRIPTION
**Pull request by xyauhideto**
#1445

Supported sites: [ccMixter](http://ccmixter.org), [TuneTrack](http://tunetrack.net/) 

Know issues:
- Unrecognised song
It's better to switch on the notification

- Beta UI
  - The player will be hidden when and after visting the home page

### CODE REVIEW

File: connectors/ccmixter.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+setupConnector();
+
+function setupConnector() {
+	if (isBetaPlayer()) {
+		setupBetaPlayer();
+	} else if (isTuneTrackPlayer()) {
+		setupTuneTrackPlayer();
+	} else {
+		setupH5Player();
```
Comment (inverse) [inline connectors/ccmixter.js]: Should we add a check for the H5 page so we can console log if we fail to match or is this generic enough to handle all cases?

File: tests/connectors/ccmixter.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+'use strict';
+
+module.exports = (driver, spec) => {
+	spec.shouldLoadWebsite(driver, {
```
Comment (inverse) [inline tests/connectors/ccmixter.js]: Is it worth adding a test for each type of player?

File: connectors/ccmixter.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+	};
+}
+
+function setupH5Player() {
+	$('audio').bind('playing pause timeupdate', Connector.onStateChanged);
+
+	Connector.getCurrentPlayer = () => {
+		let containers = document.querySelectorAll('.upload_info, .box, .trr');
+		return Array.from(containers).find((e) => {
+			let a = e.querySelector('audio');
```
Comment (inverse) [inline connectors/ccmixter.js]: Could we use more descriptive naming of variables, this isn't clear.

**Comment by xyauhideto** [comment]
Now I manage to get the flashplayer work, so I include it as well.

File: tests/connectors/ccmixter.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+	// Beta player
+	spec.shouldContainPlayerElement(driver, {
+		url: 'http://beta.ccmixter.org/tree',
+		playButtonSelector: '.play-button'
+	});
+
+	// TuneTrack player
+	spec.shouldContainPlayerElement(driver, {
+		url: 'http://tunetrack.net',
+		playButtonSelector: '.play-btn button:contains(Pause)'
```
Comment (alexesprit) [inline tests/connectors/ccmixter.js]: WebDriver throws an error when clicks on button by this selector.

File: tests/connectors/ccmixter.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+module.exports = (driver, spec) => {
+
+	// Beta player
+	spec.shouldContainPlayerElement(driver, {
```
Comment (alexesprit) [inline tests/connectors/ccmixter.js]: Is there a reason to use this function to check the connector? `shouldBehaveLikeMusicSite` works fine too for me.

**Comment by alexesprit** [comment]
Nice, thanks! 👍

**Pull request merged**